### DISCUSSION TITLE
New ideas for youtube

### DISCUSSION THREAD

**Discussion by RickzDO** [uBlockOrigin/uBlock-issues#2999] [Ideas] [answered]
Regards,

some people reported youtube now detect UBO , so far never happend to me, maybe has been updated it for this?

but i see now is many ads are getting passed, short/long ads, specially at begin of videos.. not matter which ones, i played a while to block some of them, some videos later don't load because youtube dinamically rotates their video source using diferrent googlevideos.com sites you all know this, i survive manual refreshing and finally most vids gets displayed without AD
figure out instantly when this publicity comes if you refresh the video it will load the original video without ads, at this point you guys could improve the rules creationg to have an option if instally an AD is going to be displayed, we could add automatically to do a fresh (f5) on browser to skip the AD and gets to original video been displayed, there's an option to Block and Permit but not refresh on rules creation..
to avoid doing refresh wrongly is a most to detect the real procedure which  youtube uses post/reading their servers...
using the logger before an initial AD gets displayed i usually get this:
![image](https://github.com/uBlockOrigin/uBlock-issues/assets/93270734/bd1a4ffc-47b3-485b-880c-03ace8252b22)
the top link is the AD's , maybe there are others option to fight this.. but is not a bad idea to integrate autorefresh for this cases based on blocked sites detections then do a refresh (reload), it's clean and even almost undetectable to end users
i believe this is better than having another script that is supposed to use the Skip options , when skip option gets there is too late and the video we want to see is already interrupted, also some times that approach don't work on some sceneries.. 
let me know what you guys think about..

My Best Regards

**Comment by Yuki2718** [comment, marked as answer]
Do you understand why you don't see ads when refreshed? This is all thanks to these rules:
```
youtube.com,youtubekids.com,youtube-nocookie.com##+js(set, ytInitialPlayerResponse.playerAds, undefined)
youtube.com,youtubekids.com,youtube-nocookie.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
youtube.com,youtubekids.com,youtube-nocookie.com##+js(set, ytInitialPlayerResponse.adSlots, undefined)
```
At the initial load after refreshing there's no `playerResponse` from fetch request so `ytInitialPlayerResponse` is used instead, and ads from them are addressed by the above rules while ads from `playerResponse` are currently taken care of by `trusted-replace-fetch-response` and `trusted-replace-xhr-response` rules.
> but i see now is many ads are getting passed

Reset uBO settings to default, if doing so didn't help, reinstall uBO.

**Reply by RickzDO** [reply]
never had any trouble with UBO before neither othr extension is interfering with it, so because this bombing ads from youtube lately and some how laughing at UBO i tested other option to battle this issue : hxxps://chrome.google.com/webstore/detail/skip-ad-ad-block-auto-ad/lkahpjghmdhpiojknppmlenngmpkkfma
so far haven't failed ... does this job very well

**Reply by krystian3w** [reply]
For me, it's still better to cut than to speed up the video 16 times without the skip option after 5 seconds.
